app/uk/gov/hmrc/cgtpropertydisposalsfrontend/services/returns/ReturnsService.scala

```diff
@@ -94,19 +99,70 @@ class ReturnsServiceImpl @Inject() (connector: ReturnsConnector, auditService: A
       }
     }
 
+  def deleteDraftReturns(draftReturnIds: List[UUID])(implicit hc: HeaderCarrier): EitherT[Future, Error, Unit] =
+    connector.deleteDraftReturns(draftReturnIds).subflatMap { httpResponse =>
+      if (httpResponse.status === OK)
+        Right(())
+      else
+        Left(Error(s"Call to delete draft returns came back with status ${httpResponse.status}"))
+    }
+
   def getDraftReturns(
-    cgtReference: CgtReference
+    cgtReference: CgtReference,
+    sentReturns: List[ReturnSummary]
   )(implicit hc: HeaderCarrier): EitherT[Future, Error, List[DraftReturn]] =
-    connector.getDraftReturns(cgtReference).subflatMap { httpResponse =>
-      if (httpResponse.status === OK) {
-        httpResponse
-          .parseJSON[GetDraftReturnResponse]()
-          .leftMap(Error(_))
-          .map(_.draftReturns)
-      } else {
-        Left(Error(s"Call to get draft returns came back with status ${httpResponse.status}}"))
-      }
+    for {
+      httpResponse <- connector
+                       .getDraftReturns(cgtReference)
+                       .subflatMap(r =>
+                         if (r.status === OK) Right(r)
+                         else Left(Error(s"Call to get draft returns came back with status ${r.status}}"))
+                       )
+      draftReturns <- EitherT.fromEither(
+                       httpResponse
+                         .parseJSON[GetDraftReturnResponse]()
+                         .leftMap(Error(_))
+                         .map(_.draftReturns)
+                     )
+      (sentDraftReturns, unsentDraftReturns) = draftReturns.partition(hasBeenSent(sentReturns))
+      _ <- if (sentDraftReturns.nonEmpty) EitherT.liftF[Future, Error, Unit](deleteSentDraftReturns(sentDraftReturns))
+          else EitherT.rightT[Future, Error](())
+    } yield unsentDraftReturns
+
+  private def deleteSentDraftReturns(sentDraftReturns: List[DraftReturn])(implicit hc: HeaderCarrier): Future[Unit] = {
```

Comment:
we don't care if deletes are unsuccessful when we try to delete sent returns wince we are always yielding the unsentDraftReturns in the for-comprehension so the user will not see the sent draft returns anyway

Reply:
The way the for comprehension works that submission errors always end up in the right of the EitherT in the Either. Do the type of the EitherT is `EitherT[Future, Error Either[Error, SubmitReturnResponse]]`